### PR TITLE
Task #156244 Chore: Add checkPaymentGateway method in vendor class

### PR DESCRIPTION
**Pull request by niteshkesarkar**
- Added includes folder.
- Added tjvendors class
- Added vendor class

### CODE REVIEW

File: src/com_tjvendors/site/includes/vendor.php [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function getProperties($public = true)
+	{
+		return get_object_vars($this);
+	}
+
+	/**
+	 * Method to get gateway details
```
Comment (coolbung) [inline src/com_tjvendors/site/includes/vendor.php]: Please make sure the comment correctly describes the function

Comment (niteshkesarkar) [reply src/com_tjvendors/site/includes/vendor.php]: @coolbung  - Described method properly

File: src/com_tjvendors/site/includes/vendor.php [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+	 *
+	 * @return  boolean  True if empty false otherwise
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function checkPaymentGateway()
```
Comment (thite-amol) [inline src/com_tjvendors/site/includes/vendor.php]: @niteshkesarkar change name to `getPyamentConfig` or `getGatewatDetails`

cc
@coolbung

File: src/com_tjvendors/site/includes/vendor.php [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+	{
+		// Check for the client in the xref
+		$xreftable = TJVendors::table("vendorclientxref");
+
+		if (!$xreftable->load(array('vendor_id' => $this->vendor_id, 'client' => $this->client)))
+		{
+			return false;
+		}
+
+		$params = json_decode($xreftable->params);
```
Comment (thite-amol) [inline src/com_tjvendors/site/includes/vendor.php]: @niteshkesarkar params already merged in the vendor object in the `setClient` method read the details from the object itself.

File: src/com_tjvendors/site/includes/vendor.php (hunk 2 above)
Comment (niteshkesarkar) [reply src/com_tjvendors/site/includes/vendor.php]: @coolbung @thite-amol - Used  `getPaymentConfig` as a method name.

File: src/com_tjvendors/site/includes/vendor.php (hunk 3 above)
Comment (niteshkesarkar) [reply src/com_tjvendors/site/includes/vendor.php]: @thite-amol - Used value from `$this->payment_gateway` property.

**Comment by manojLondhe** [comment]
@niteshkesarkar rebase these changes on 1.4.0 branch

**Comment by manojLondhe** [comment]
@thite-amol do first review and add `approve review` comment

**Comment by thite-amol** [comment]
@manojLondhe Please accept this PR

**Pull request merged**